bundles/org.openhab.binding.benqprojector/src/main/java/org/openhab/binding/benqprojector/internal/discovery/MulticastListener.java

```diff
@@ -57,20 +61,25 @@ public class MulticastListener {
      */
     public MulticastListener(String ipv4Address) throws IOException, SocketException {
         InetAddress ifAddress = InetAddress.getByName(ipv4Address);
-        NetworkInterface netIF = NetworkInterface.getByInetAddress(ifAddress);
+        NetworkInterface networkInterface = getMulticastInterface(ipv4Address);
         logger.debug("Discovery job using address {} on network interface {}", ifAddress.getHostAddress(),
-                netIF != null ? netIF.getName() : "UNKNOWN");
+                networkInterface.getName());
         socket = new MulticastSocket(AMX_MULTICAST_PORT);
-        socket.setInterface(ifAddress);
+        socket.setNetworkInterface(networkInterface);
         socket.setSoTimeout(DEFAULT_SOCKET_TIMEOUT_SEC);
-        InetAddress mcastAddress = InetAddress.getByName(AMX_MULTICAST_GROUP);
-        socket.joinGroup(mcastAddress);
+        inetSocketAddress = new InetSocketAddress(InetAddress.getByName(AMX_MULTICAST_GROUP), AMX_MULTICAST_PORT);
+        socket.joinGroup(inetSocketAddress, null);
         logger.debug("Multicast listener joined multicast group {}:{}", AMX_MULTICAST_GROUP, AMX_MULTICAST_PORT);
     }
 
     public void shutdown() {
         logger.debug("Multicast listener closing down multicast socket");
-        socket.close();
+        try {
+            socket.leaveGroup(inetSocketAddress, null);
+            socket.close();
+        } catch (IOException e) {
+            logger.debug("Exception shutting down multicast socket");
```

Comment:
For this to be a bit more useful, perhaps add the exception message?
```suggestion
            logger.debug("Exception shutting down multicast socket: {}", e.getMessage());
```

(applies to the other two similar places in the PR as well)

Reply:
Good catch. Fixed.